src/main/java/com/nicusa/controller/DrugController.java

```diff
@@ -73,6 +75,70 @@
     return new ResponseEntity<>(drugAssembler.toResource(drug), HttpStatus.OK);
   }
 
+  @Transactional
+  @ResponseBody
+  @RequestMapping(value = "/portfolio/drug", method = RequestMethod.POST, consumes = "application/json")
+  public ResponseEntity<?> saveDrug(@AuthenticationPrincipal Principal principal,
+                                    @RequestBody DrugResource drugResource) {
+    Long loggedInUserProfileId = securityController.getAuthenticatedUserProfileId();
+    if (principal != null && ((UsernamePasswordAuthenticationToken)principal).getPrincipal() == loggedInUserProfileId) {
+      UserProfile userProfile = entityManager.find(UserProfile.class, loggedInUserProfileId);
+      if (userProfile == null) {
+        return new ResponseEntity<>(HttpStatus.NOT_FOUND);
+      } else {
+        Drug drug = drugResourceToDomainConverter.convert(drugResource);
+        entityManager.persist(drug);
+        Portfolio portfolio = userProfile.getPortfolio();
```

Comment:
We should not do this in one call.  The drug controller is for managing drug resources not profile resources.  Please remove.  This should be done on the client side.